tests/pytests/functional/states/test_x509_v2.py

```diff
@@ -0,0 +1,2638 @@
+    indirect=True,
+)
+@pytest.mark.skipif(
+    CRYPTOGRAPHY_VERSION[0] < 37,
+    reason="Parsing of RFC4514 strings requires cryptography >= 37",
+)
+def test_certificate_managed_with_signing_policy_no_subject_override_no_changes(
+    x509, cert_args
+):
+    cert_args["subject"] = "CN=from_call"
+    ret = x509.certificate_managed(**cert_args)
+    _assert_not_changed(ret)
+
+
+@pytest.mark.usefixtures("existing_cert_chain")
+@pytest.mark.parametrize(
+    "existing_cert_chain",
+    [
+        {"encoding": "pem"},
+        pytest.param(
+            {"encoding": "pkcs12"},
+            marks=pytest.mark.skipif(
+                CRYPTOGRAPHY_VERSION[0] < 36,
+                reason="PKCS12 serialization requires cryptography v36+",
+            ),
+        ),
+        pytest.param(
+            {"encoding": "pkcs7_pem"},
+            marks=pytest.mark.skipif(
+                CRYPTOGRAPHY_VERSION[0] < 37,
+                reason="PKCS7 serialization requires cryptography v37+",
+            ),
+        ),
+        pytest.param(
+            {"encoding": "pkcs7_der"},
+            marks=pytest.mark.skipif(
+                CRYPTOGRAPHY_VERSION[0] < 37,
+                reason="PKCS7 serialization requires cryptography v37+",
+            ),
+        ),
+    ],
+    indirect=True,
+)
+def test_certificate_managed_existing_chain(x509, cert_args):
+    ret = x509.certificate_managed(**cert_args)
+    _assert_not_changed(ret)
+
+
+@pytest.mark.usefixtures("existing_cert")
+@pytest.mark.parametrize(
+    "existing_cert",
+    [
+        {"encoding": "pkcs12"},
+        {"encoding": "pkcs12", "pkcs12_friendlyname": "littlefinger"},
+        {"encoding": "pkcs12", "pkcs12_passphrase": "p4ssw0rd"},
+    ],
+    indirect=True,
+)
+@pytest.mark.skipif(
+    CRYPTOGRAPHY_VERSION[0] < 36,
+    reason="PKCS12 serialization requires cryptography v36+",
+)
+def test_certificate_managed_existing_pkcs12(x509, cert_args):
+    """
+    Ensure PKCS12 encoding does not lead to reissued certificates in other ways.
+    pkcs12_compat is not checked currently
+    """
+    ret = x509.certificate_managed(**cert_args)
+    _assert_not_changed(ret)
+
+
+@pytest.mark.usefixtures("existing_file")
+def test_certificate_managed_existing_not_a_cert(x509, cert_args, rsa_privkey, ca_key):
+    """
+    If `name` is not a valid certificate, a new one should be issued at the path
+    """
+    cert_args["private_key"] = rsa_privkey
+    ret = x509.certificate_managed(**cert_args)
+    _assert_cert_created_basic(ret, cert_args["name"], rsa_privkey, ca_key)
+
+
+@pytest.mark.usefixtures("existing_cert")
+@pytest.mark.parametrize("days,expected", [(1, False), (9999, True)])
+def test_certificate_managed_days_remaining(x509, cert_args, days, expected):
+    """
+    The certificate should be reissued if days_remaining indicates it
+    """
+    cert_args["days_remaining"] = days
+    ret = x509.certificate_managed(**cert_args)
+    assert bool(ret.changes) == expected
```

Comment:
```suggestion
    assert bool(ret.changes) is expected
```
`expected` is a boolean, as such, `is` should be used.